source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/QueryServices/ChargePointQueryServiceTests.cs

```diff
@@ -0,0 +1,318 @@
+            var unexpectedPoint = new Point(10.00m, InstantHelper.GetTodayPlusDaysAtMidnightUtc(2));
+            var expectedPoints = new List<Point> { expectedPoint, unexpectedPoint };
+
+            var expectedCharge = await GetValidCharge(chargesDatabaseWriteContext, expectedPoints, Resolution.PT15M);
+
+            chargesDatabaseWriteContext.Charges.Add(expectedCharge);
+            await chargesDatabaseWriteContext.SaveChangesAsync();
+
+            await using var chargesDatabaseQueryContext = _databaseManager.CreateDbQueryContext();
+            var sut = GetSut(chargesDatabaseQueryContext);
+            var searchCriteria = new ChargePriceSearchCriteriaV1Dto(
+                expectedCharge.Id,
+                InstantHelper.GetTodayAtMidnightUtc().ToDateTimeUtc(),
+                InstantHelper.GetTomorrowAtMidnightUtc().ToDateTimeUtc());
+
+            var expected = new ChargePriceV1Dto(
+                expectedPoint.Price,
+                InstantHelper.GetTodayPlusMinutesAtMidnightUtc(0).ToDateTimeOffset(),
+                InstantHelper.GetTodayPlusMinutesAtMidnightUtc(15).ToDateTimeOffset());
+
+            // Act
+            var actual = sut.Search(searchCriteria);
+
+            // Assert
+            actual.Should().HaveCount(1);
+            actual.Single().Should().Be(expected);
+        }
+
+        [Fact]
+        public async Task SearchAsync_WhenSearching_ReturnsPricesWith15MinutesInterval()
+        {
+            // Arrange
+            await using var chargesDatabaseWriteContext = _databaseManager.CreateDbContext();
+
+            var points = new List<Point>
+            {
+                new(1.00m, InstantHelper.GetTodayPlusMinutesAtMidnightUtc(0)),
+                new(2.00m, InstantHelper.GetTodayPlusMinutesAtMidnightUtc(15)),
+                new(3.00m, InstantHelper.GetTodayPlusMinutesAtMidnightUtc(30)),
+                new(4.00m, InstantHelper.GetTodayPlusMinutesAtMidnightUtc(45)),
+                new(5.00m, InstantHelper.GetTodayPlusMinutesAtMidnightUtc(60)),
+            };
+
+            var charge = await GetValidCharge(chargesDatabaseWriteContext, points, Resolution.PT15M);
+
+            chargesDatabaseWriteContext.Charges.Add(charge);
+            await chargesDatabaseWriteContext.SaveChangesAsync();
+
+            await using var chargesDatabaseQueryContext = _databaseManager.CreateDbQueryContext();
+            var sut = GetSut(chargesDatabaseQueryContext);
+            var searchCriteria = new ChargePriceSearchCriteriaV1Dto(
+                charge.Id,
+                InstantHelper.GetTodayAtMidnightUtc().ToDateTimeUtc(),
+                InstantHelper.GetTomorrowAtMidnightUtc().ToDateTimeUtc());
+
+            // Act
+            var actual = sut.Search(searchCriteria);
+
+            // Assert
+            actual.Should().HaveCount(points.Count);
+            Assert.True(actual.All(a => (a.ActiveToDateTime - a.ActiveFromDateTime).Minutes == 15));
```

Comment:
If possible, use fluent assertions
```suggestion
            actual.All(a => (a.ActiveToDateTime - a.ActiveFromDateTime).Minutes.Should().Be(15));
```